Add ADF readme (#592)
MetaphorData/connectors: metaphor/azure_data_factory/readme.md

# Azure Data Factory Connector

This connector extracts technical metadata from Azure Data Factories using [Azure REST API](https://learn.microsoft.com/en-us/rest/api/datafactory/v2)

## Setup

We recommend creating a dedicated Azure AD Application for the connector to use.

1. Follow [Step 1 of this doc](https://docs.microsoft.com/en-us/power-bi/developer/embedded/embed-service-principal#step-1---create-an-azure-ad-app) to create an Azure AD app and a client secret.

> 2. Give the app a reader role to access REST APIs. Go to the resource you want to connect to. For example, a subscription. On the resource page, select `Access Control (IAM)`, and add the built-in `Reader` role to the app you created in Step 1.

Reviewer: I'm not sure I follow this completely. So is the idea to give global READ permission to the subscription the ADF belongs to? Can we do it at a finer grain?

Author: Yes, you're right. It's a global reader role. There are no built-in read-only role for ADF, we can provide a role permission in the document as well.

> 2. Give the app the [Reader Role](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#reader) for each data factory to access the REST APIs. Select `Access Control (IAM)` from each data factory's settings page and add the built-in `Reader` role to the app you created in Step 1.

Reviewer: ```suggestion
2. Grant the app the [Reader Role](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#reader) to each data factory. Select `Access Control (IAM)` from each data factory's settings page and add the built-in `Reader` role to the app you created in Step 1.
```

## Config File

Create a YAML config file based on the following template.

### Required Configurations

```yaml
tenant_id: <tenant_id>  # The Azure Directory (tenant) ID

client_id: <client_id>  # The Azure Application client id

client_secret: <client_secret>  # The client secret value (not secret ID)

> subscription_id: <subscription_id>  # Azure subscription id

Reviewer: What if there are multiple Azure data factories using different subscription IDs? Do we need to create one for each subscription?

Author: Yes, we will need to create one for each subscription.

1. Follow [Step 1 of this doc](https://docs.microsoft.com/en-us/power-bi/developer/embedded/embed-service-principal#step-1---create-an-azure-ad-app) to create an Azure AD app and a client secret.

> 2. Give the app a reader role for each data factory to access REST APIs. In the Azure portal, search `Data Factories`, click the data factory you want to connect to. Select `Access Control (IAM)`, and add the built-in `Reader` role to the app you created in Step 1.

Reviewer: ```suggestion
2. Give the app the [Reader Role](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#reader) for each data factory to access the REST APIs. Select `Access Control (IAM)` from each data factory's settings page and add the built-in `Reader` role to the app you created in Step 1.
```
